### PR TITLE
[BEAM-13993] [BEAM-10039] Import beam plugins before starting SdkHarness

### PR DESCRIPTION
**Pull request by rahuliyer95**
The `--beam_plugin` `PipelineOption` is current unused and does not import plugins before starting the `SdkHarness`. This PR aims to import the plugins before starting the `SdkHarness`.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/validate-runner/README.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by rahuliyer95** [comment]
R: @reuvenlax @TheNeuralBit

**Comment by TheNeuralBit** [comment]
Hm I'm not familiar with the history of the `--beam_plugin` argument. Maybe @tvalentyn or @aaltay know more?

Did this break at some point in the transition to portability?

**Comment by tvalentyn** [comment]
This looks to be one of the lesser known features of Dataflow runner V1, and is marked experimental. I wasn't aware of it. @rahuliyer95 would you be interested in making it a non-experimental first-class citizen in Beam that would work for all runners? I would recommend an advertisement on dev describing your use-cases for this, settle on the expected plugin format (class vs module to remove the ambiguity or allow both if that makes sense sometimes - if so you could add rationale), adding unit tests and proper documentation, removing the 'experimental' status from the doc. You can start the conversation on dev@ with a proposal to remove the 'experimental' bit and reference your PR.

This is what the import code looks like in the Dataflow Python worker:

```
  for beam_plugin in sdk_pipeline_options.get('beam_plugins', []):
    try:
      module_name, class_name = beam_plugin.rsplit('.', 1)
      _ = getattr(importlib.import_module(module_name), class_name)
      logging.info('Successfully imported beam plugin %s', beam_plugin)
    except ImportError:
      logging.warning('Failed to import beam plugin %s', beam_plugin,
                      exc_info=True)
```

**Comment by tvalentyn** [comment]
cc: @ryanthompson591

File: sdks/python/apache_beam/runners/worker/sdk_worker_main.py [hunk 1]
```diff
@@ -48,6 +49,31 @@
...
+            "Looks like %s is not a module. "
+            "Trying to import it assuming it's a class"
+          ),
+          plugin
+        )
+        module, _ = plugin.rsplit('.', 1)
+        importlib.import_module(module)
+        _LOGGER.info('Imported %s for beam-plugin %s', module, plugin)
+      except ImportError:
+        _LOGGER.warning(
```
Comment (robertwb) [inline sdks/python/apache_beam/runners/worker/sdk_worker_main.py:70]: Should this instead be a fatal error?

Comment (rahuliyer95) [reply sdks/python/apache_beam/runners/worker/sdk_worker_main.py:70]: I feel if a plugin isn't imported we should continue with the pipeline execution, would love to hear your rationale about why would we want to make this fatal?

**Pull request closed without merging**